controllers/openstackdataplane_controller.go

```diff
@@ -141,59 +130,58 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
+			return ctrl.Result{RequeueAfter: time.Second * 5}, nil
 		}
 
-		for _, role := range roles.Items {
-			logger.Info("DataPlane deploy", "role.Name", role.Name)
-			if role.Spec.DataPlane != instance.Name {
-				err = fmt.Errorf("role %s: role.DataPlane does not match with role.Label", role.Name)
-				deployErrors = append(deployErrors, "role.Name: "+role.Name+" error: "+err.Error())
-			}
-			logger.Info("Role", "DeployStrategy.Deploy", role.Spec.DeployStrategy.Deploy, "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-			if !role.Spec.DeployStrategy.Deploy {
-				_, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), &role, func() error {
-					logger.Info("Reconciling Role", "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-					helper.GetLogger().Info("CreateOrPatch Role.DeployStrategy.Deploy", "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-					role.Spec.DeployStrategy.Deploy = instance.Spec.DeployStrategy.Deploy
-					if err != nil {
-						deployErrors = append(deployErrors, "role.Name: "+role.Name+" error: "+err.Error())
-					}
-					return nil
-				})
-				if err != nil {
-					deployErrors = append(deployErrors, "role.Name: "+role.Name+" error: "+err.Error())
+		for _, nodeSet := range nodeSets.Items {
+			logger.Info("DataPlane deploy", "nodeSet.Name", nodeSet.Name)
+			// We don't expect that our nodeSet list will be overly large. As such, linear search is likely
+			// to be computationally more efficient compared to importing and using slices.Contains() here.
+			nodeSetFound := false
+			for _, nodeSetName := range instance.Spec.NodeSets {
+				if nodeSetName == nodeSet.Name {
+					nodeSetFound = true
 				}
 			}
-			if !role.IsReady() {
-				logger.Info("Role", "IsReady", role.IsReady(), "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
+
+			if !nodeSetFound {
+				err = fmt.Errorf("nodeSet %s: nodeSet.DataPlane does not match with nodeSet.Label", nodeSet.Name)
+				deployErrors = append(deployErrors, "nodeSet.Name: "+nodeSet.Name+" error: "+err.Error())
+			}
+
+			if !nodeSet.IsReady() {
+				logger.Info("NodeSet", "IsReady", nodeSet.IsReady(), "NodeSet.Namespace", instance.Namespace, "NodeSet.Name", nodeSet.Name)
 				shouldRequeue = true
-				mirroredCondition := role.Status.Conditions.Mirror(condition.ReadyCondition)
+				mirroredCondition := nodeSet.Status.Conditions.Mirror(condition.ReadyCondition)
 				if mirroredCondition != nil {
-					logger.Info("Role", "Status", mirroredCondition.Message, "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
+					logger.Info("NodeSet", "Status", mirroredCondition.Message, "NodeSet.Namespace", instance.Namespace, "NodeSet.Name", nodeSet.Name)
 					if condition.IsError(mirroredCondition) {
-						deployErrors = append(deployErrors, "role.Name: "+role.Name+" error: "+mirroredCondition.Message)
+						deployErrors = append(deployErrors, "nodeSet.Name: "+nodeSet.Name+" error "+mirroredCondition.Message)
```

Comment:
Alright, so for the time being, the way to deploy a DataPlane with all NodeSets is to individually set DeployStrategy.Deploy=true on each NodeSet?

Reply:
Yep, correct. There's no other mechanism here atm.